tslearn/metrics.py

```diff
@@ -654,6 +655,85 @@ def cdist_dtw(dataset1, dataset2=None, global_constraint=None,
     """
     dataset1 = to_time_series_dataset(dataset1)
 
+    if dataset2 is None:
+        # Inspired from code by @GillesVandewiele:
+        # https://github.com/rtavenar/tslearn/pull/128#discussion_r314978479
+        matrix = numpy.zeros((len(dataset1), len(dataset1)))
```

Comment:
One nice unit test for this could be to assert that `cdist_dtw(dataset1, dataset1)` is equal to `cdist_dtw(dataset1)`